src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -376,12 +379,50 @@ protected SCMRevision doRetrieve(SCMHead head, TaskListener listener, GHReposito
         if (head instanceof PullRequestSCMHead) {
             int number = ((PullRequestSCMHead) head).getNumber();
             ref = repo.getRef("pull/" + number + "/merge");
+            // TODO if we already had the GHPullRequest.user.login in a field in the PullRequestSCMHead, we could pass that directly and save an API call
+            String trustedBase = trustedReplacement(repo, repo.getPullRequest(number));
+            if (trustedBase != null) {
+                return new UntrustedPullRequestSCMRevision(head, ref.getObject().getSha(), trustedBase);
+            }
         } else {
             ref = repo.getRef("heads/" + head.getName());
         }
         return new SCMRevisionImpl(head, ref.getObject().getSha());
     }
 
+    @Override
+    public SCMRevision getTrustedRevision(SCMRevision revision, TaskListener listener) throws IOException, InterruptedException {
+        if (revision instanceof UntrustedPullRequestSCMRevision) {
+            PullRequestSCMHead head = (PullRequestSCMHead) revision.getHead();
+            UntrustedPullRequestSCMRevision rev = (UntrustedPullRequestSCMRevision) revision;
+            listener.getLogger().println("Loading trusted files from target branch at " + rev.baseHash + " rather than " + rev.getHash());
```

Comment:
Note that `rev.getHash()` here is actually the synthetic merge commit, which is confusing. When I fix this branch source to store precise information in its `SCMRevision`s, this will need to be rewritten to use two kinds of revisions:

* origin-branch revisions, consisting simply of a commit hash (`SCMRevisionImpl` works for this purpose)
* PR revisions, consisting of a PR head commit plus a base branch commit, and a trusted flag

`getTrustedRevision` would convert an untrusted PR revision into an origin revision using the base commit, but leave other revisions alone. `build` on a PR revision would check out the base commit and merge the PR head commit.